mlir/utils/performance/quickTunerGen.py

```diff
@@ -0,0 +1,1115 @@
+                feature = tuple(row[:-1])
+                if feature not in added_features:
+                    added_features.add(feature)
+                    final_dataset.append(feature)
+                    if len(final_dataset) >= self.N:
+                        break
+            # add more high performers
+        
+            
+        return pd.DataFrame(final_dataset, columns=cols).head(self.N) # though this should really not be set
+
+    def getConfig(self, combined_df):
+        config_dict = {}
+    
+        type_gemm_dict = orderByGemmType(combined_df, normalize=self.normalize)
+
+        type_gemm_dict = orderGemmDict(type_gemm_dict)
+
+        for dtype, dfs in type_gemm_dict.items():
+            
+            top_90_percent = []
+            for cfg in dfs:
+                df = dfs[cfg]
+                top_90_percent.append(self.__get_top_90_percent(df))
+
+            config_dict[dtype] = self.__build_final_df(top_90_percent)
+        
+        self.config = config_dict
+        return config_dict
+
+
+
+def main(args=None):
+    if args is None:
+        args = sys.argv[1:]
+
+    parser = argparse.ArgumentParser(prog='clusterConfigs.py',
+                                     description='Bunch together runs into a parallel dir named ../{DIR_NAME}-bunch')
+
+    parser.add_argument('--input-file',
+                        required=True,
+                        type=str)
+
+    parser.add_argument('--method',
+                        nargs='+',
+                        choices=["default","topNSelect","topMode","takeNEach","fairSelect","hardcoded"],
+                        default=["default","fairSelect"],
+                        help='Select perfConfig gen selection method')
+
+    parser.add_argument('--save',
+                        action='store_true',
+                        default=False,
+                        help='Save configs to name.dtype.qt')
+
+    parser.add_argument('--debug',
+                        action='store_true',
+                        default=False,
+                        help='Print debug info, print config files to stdout')
+
+    parser.add_argument('--num', '-n',
+                        type=int,
+                        default=40,
+                        help='Number of perf configs to include')
+
+    parser.add_argument('--perfconfig-format',
+                        action='store_true',
+                        default=False,
+                        help='Save file in correct csv perfconfig format')
+
+    parser.add_argument('--directory',
+                        type=str,
+                        help='Directory to store results to')
+
+    pargs = parser.parse_args()
+
+    tuner = quickTuner(pargs)
+
+    tuner.tune()
+
+    if pargs.save:
+        tuner.saveConfigs(pf_format=pargs.perfconfig_format)
+
+    if pargs.debug:
+        tuner.printConfigs()
+            
+if __name__ == '__main__':
+    main(sys.argv[1:])
+
```

Comment:
Something feels really awkward here - I can't for the life of me follow what all this is supposed to be doing and there's a whole lot of for loops and manual bouncing around of columns